### PR TITLE
Add shares transfer

### PR DESCRIPTION
**Pull request by Index0011**
Add shares transfer

### CODE REVIEW

File: pallets/phala/src/compute/base_pool.rs [hunk 1]
```diff
@@ -575,6 +577,59 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		#[pallet::weight(0)]
+		#[frame_support::transactional]
+		pub fn transfer_shares_arbitrary(
```
Comment (h4x3rotab) [inline pallets/phala/src/compute/base_pool.rs:583]:
```suggestion
		pub fn backfill_transfer_shares(
```

File: pallets/phala/src/compute/base_pool.rs [hunk 2]
```diff
@@ -575,6 +577,59 @@ pub mod pallet {
...
+				let mut dest_nft_guard = Self::get_nft_attr_guard(
+					base_pool_info.cid,
+					dest_nft_id,
+				)
+				.expect("get nftattr should always success; qed.");
+				
+				ensure!(from_nft_guard.attr.shares >= *shares, Error::<T>::TransferSharesAmountInvalid);
+				from_nft_guard.attr.shares -= *shares;
+				dest_nft_guard.attr.shares += *shares;
+				from_nft_guard.save()?;
```
Comment (h4x3rotab) [inline pallets/phala/src/compute/base_pool.rs:628]: Will it emit events for SubSquid to process?

File: pallets/phala/src/compute/base_pool.rs (hunk 1 above)
Comment (Index0011) [reply pallets/phala/src/compute/base_pool.rs]: done

File: pallets/phala/src/compute/base_pool.rs (hunk 2 above)
Comment (Index0011) [reply pallets/phala/src/compute/base_pool.rs]: use burn and mint to adjust subsquid

**Review by h4x3rotab — APPROVED** [review]

**Pull request merged**